utility/src/test/java/org/oppia/android/util/networking/NetworkConnectionTestUtilTest.kt

```diff
@@ -55,56 +57,82 @@ class NetworkConnectionTestUtilTest {
-    setNetworkConnectionStatus(
+  fun testSetNetworkInfo_wifiShadowNetwork_notConnected_returnsCorrectNetworkInfo() {
+    networkConnectionTestUtil.setNetworkInfo(
       status = ConnectivityManager.TYPE_WIFI,
       networkState = NetworkInfo.State.DISCONNECTED
     )
-    assertThat(networkConnectionUtilProdImpl.getCurrentConnectionStatus()).isEqualTo(NONE)
+    assertThat(
+      checkForCorrectNetworkConnection(
+        status = ConnectivityManager.TYPE_WIFI,
+        state = NetworkInfo.State.DISCONNECTED
+      )
+    ).isTrue()
   }
 
   @Test
-  fun testSetNetworkInfo_cellularShadowNetwork_connected_connectionStatusIsCellular() {
-    setNetworkConnectionStatus(
+  fun testSetNetworkInfo_cellularShadowNetwork_connected_returnsCorrectNetworkInfo() {
+    networkConnectionTestUtil.setNetworkInfo(
       status = ConnectivityManager.TYPE_MOBILE,
       networkState = NetworkInfo.State.CONNECTED
     )
-    assertThat(networkConnectionUtilProdImpl.getCurrentConnectionStatus()).isEqualTo(CELLULAR)
+    assertThat(
+      checkForCorrectNetworkConnection(
+        status = ConnectivityManager.TYPE_MOBILE,
+        state = NetworkInfo.State.CONNECTED
+      )
+    ).isTrue()
   }
 
   @Test
-  fun testSetNetworkInfo_cellularShadowNetwork_notConnected_connectionStatusIsNone() {
-    setNetworkConnectionStatus(
+  fun testSetNetworkInfo_cellularShadowNetwork_notConnected_returnsCorrectNetworkInfo() {
+    networkConnectionTestUtil.setNetworkInfo(
       status = ConnectivityManager.TYPE_MOBILE,
       networkState = NetworkInfo.State.DISCONNECTED
     )
-    assertThat(networkConnectionUtilProdImpl.getCurrentConnectionStatus()).isEqualTo(NONE)
+    assertThat(
+      checkForCorrectNetworkConnection(
+        status = ConnectivityManager.TYPE_MOBILE,
+        state = NetworkInfo.State.DISCONNECTED
+      )
+    ).isTrue()
   }
 
   @Test
-  fun testSetNetworkInfo_noActiveShadowNetwork_connectionStatusIsNone() {
-    setNetworkConnectionStatus(
+  fun testSetNetworkInfo_noActiveShadowNetwork_returnsCorrectNetworkInfo() {
+    networkConnectionTestUtil.setNetworkInfo(
       status = NO_CONNECTION,
       networkState = NetworkInfo.State.DISCONNECTED
     )
-    assertThat(networkConnectionUtilProdImpl.getCurrentConnectionStatus()).isEqualTo(NONE)
+    assertThat(
+      checkForCorrectNetworkConnection(
+        status = NO_CONNECTION,
+        state = NetworkInfo.State.DISCONNECTED
+      )
+    ).isTrue()
   }
 
-  private fun setNetworkConnectionStatus(status: Int, networkState: NetworkInfo.State) {
-    NetworkConnectionTestUtil.setNetworkInfo(context, status, networkState)
+  private fun checkForCorrectNetworkConnection(status: Int, state: NetworkInfo.State): Boolean {
+    val checkType = connectivityManager.activeNetworkInfo?.type == status
+    val checkState = connectivityManager.activeNetworkInfo?.state == state
+    return checkType && checkState
   }
```

Comment:
This might actually be reducing readability by hiding some of the context. Consider instead doing something like:

```kotlin
assertThat(connectivityManager.getActiveType()).isEqualTo(NO_CONNECTION)
assertThat(connectivityManager.getActiveState()).isEqualTo(NetworkInfo.State.DISCONNECTED)

...

private fun ConnectivityManager.getActiveType() = activeNetworkInfo?.type
private fun ConnectivityManager.getActiveState() = activeNetworkInfo?.state
```

That might help improve test readability a bit since it's making it much clearer what is being checked. And, if a check fails, the error will probably be much more clear (rather than "is not true" which would be the case for the current code).

Reply:
Done.